data/fileio/CSVStreamWriter.h

```diff
@@ -0,0 +1,100 @@
+/* -*- c-basic-offset: 4 indent-tabs-mode: nil -*-  vi:set ts=8 sts=4 sw=4: */
+
+/*
+    Sonic Visualiser
+    An audio file viewer and annotation editor.
+    Centre for Digital Music, Queen Mary, University of London.
+    This file copyright 2017 Lucas Thompson.
+    
+    This program is free software; you can redistribute it and/or
+    modify it under the terms of the GNU General Public License as
+    published by the Free Software Foundation; either version 2 of the
+    License, or (at your option) any later version.  See the file
+    COPYING included with this distribution for more information.
+*/
+
+#ifndef _CSV_STREAM_WRITER_H_
+#define _CSV_STREAM_WRITER_H_
```

Comment:
You wouldn't know this (don't you love review comments that begin like that) but the guard symbols of form `_THIS_N_THAT_H_` are vestigial from before I realised that a leading underscore was supposed to indicate a system-reserved symbol. More recent code uses `SV_THIS_N_THAT_H`. I've never done a clean replacement sweep, just updated them as and when I happen to be working in a file.